src/grisp_gpio.erl

```diff
@@ -1,3 +1,24 @@
+%% -----------------------------------------------------------------------------
+%% @doc GRiSP API for configuration of pins
+%%
+%% @type pin() = integer() | Pin
```

Comment:
Wouldn't it make sense to have this as a proper typespec instead of a Edoc comment?
I think it would be picked up by edoc as well, wouldn't it?

Same goes for the other `@type` or `@spec`

Reply:
For the `@spec` I often find the doc version more readable, but I think one can have both then. Similar argument for the `@type`. Correct me if I'm wrong, but here we would either have an extra type or all atoms and `integer()` in one line when we use the proper typespec.

Reply:
Official documentation recommends `-spec` attributes since they can be used by other tools (and should be identical anyways): http://erlang.org/doc/apps/edoc/chapter.html#Type_specifications

Reply:
I tried out the different possibilities:

The current version with `@type` looks like the following in my browser:
<img width="1302" alt="Screenshot using edoc comment" src="https://user-images.githubusercontent.com/6095801/73118829-dd8f0b80-3f59-11ea-88d3-34d1c9ac79b6.png">

Using `-type` 
```suggestion1
-type pin() :: integer() | gpio1_1 | gpio1_2 | gpio1_3 | gpio1_4
         | gpio2_1 | gpio2_2 | gpio2_3 | gpio2_4
         | led1_r | led1_g | led1_b | led2_r | led2_g | led2_b
         | jumper_1 | jumper_2 | jumper_3 | jumper_4 | jumper_5
         | spi1_pin7 | spi1_pin8 | spi1_pin9 | spi1_pin10
         | ss1 | ss2
         | uart_1_cts | uart_2_txd | uart_3_rxd | uart_4_rts.
-export_type([pin/0]).         
```
looks like
<img width="1310" alt="Screenshot using type spec" src="https://user-images.githubusercontent.com/6095801/73118863-5d1cda80-3f5a-11ea-81df-5d4edca90f21.png">

I couldn't find a reference, but it seems that one needs to export the type to generate the documentation from a type spec with `-type`.  I'm not sure whether we want that for all types.

One can combine both approaches, if there is a type spec with `-type` and an edoc comment with `@type` edoc will use the comment. Maybe that's a good solution, for types we don't want to export or if we want to improve the readability of the documentation (like in this case here).

Reply:
I’d stick to the specs since those are universally supported. Having duplicate definitions will only lead to code rot on the long run I think.

Reply:
The only appearing exported types was a conundrum.  There is no doc about this except in the Edoc Release notes:

http://erlang.org/doc/apps/edoc/notes.html#edoc-0.7.7

> Not all Erlang types are included in the documentation, but only those exported by some export_type declaration or used by some documented Erlang specification (-spec).

That might be a nice PR to Edocs and/or Typespecs documentation